ocm_python_wrapper/cluster.py

```diff
@@ -257,33 +258,112 @@ def addon_info(self):
             self.addon_name
         ).to_dict()
 
-    def validate_addon_parameters(self, parameters):
-        _info = self.addon_info()
-        _parameters = _info.get("parameters")
-        if not _parameters and parameters:
+    def validate_addon_parameters(self, user_parameters, use_api_defaults=True):
+        """Checks user input for addon parameters.
+
+        Args:
+            user_parameters (List) : User parameter input set with values.
+            use_api_defaults (bool) : If true, set required parameter with default value to not fail as missing.
+
+        Returns: Updated parameters (if default values updated) to provide for installation.
+
+        Raises:
+            ValueError: When a required parameter is missing,
+                        or when parameters are passed but not needed for addon.
+        """
+
+        def _get_required_cluster_parameters(parameters):
+            """Uses documentation of the addon parameters and filters them by conditions related to the cluster.
+
+            Args:
+                parameters (dict) : Addons parameters from api.
+
+            Returns:
+                Dict of parameters corresponding to the clusters the current configuration and
+                              required for installation.
+
+            Examples:
+                parameters = {
+                'items': [{'id': 'rosa-cli-required', 'name': 'ROSA CLI Required', 'description':
+            '...', 'value_type': 'boolean', 'validation': '^true$', 'validation_err_msg': '', 'required': True,
+            'editable': False, 'enabled': True, 'default_value': '', 'editable_direction': '', 'conditions': [{
+            'resource': 'cluster', 'data': {'aws.sts.enabled': True, 'cloud_provider.id': 'aws'}}]}, ...]
+            }
```

Comment:
this is what the api returns, no need to provide an example

Reply:
This is an example of the input `parameters` (of the specific addon) provided from the api with `required`, `conditions` etc.